### ISSUE TITLE
Amplify AppSync results cached

### ISSUE THREAD

**Issue by essedbl** [aws-amplify/amplify-hosting#3528]
### Before opening, please confirm:

- [X] I have checked to see if my question is addressed in the [FAQ](https://github.com/aws-amplify/amplify-hosting/blob/master/FAQ.md).
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-hosting/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting questions](https://github.com/aws-amplify/amplify-hosting/blob/master/CONTRIBUTING.md).
- [X] I have removed any sensitive information from my code snippets and submission.

### App Id

d3rjzy76ilz45f

### AWS Region

us-west-2

### Amplify Hosting feature

Not Applicable

### Question

I setup a website using AWS Amplify Hosting. The front-end is NextJS. I am using the API Category in the Amplify Backend, which is in turn is using AWS App Sync and DynamoDB. **Caching is set to None, in the AppSync API.**

I am using NextJS, because I need to support some server-side rendering (different feature/topic). It happens that I am calling the API.query operation from within the `getServerSideProps()`, and somewhere, somehow, data is being cached and I cannot clear the cache.

Using the amplify studio (website), I navigate to the Content section, and delete records. Even after several days, the records I deleted are still returned by my API.query calls. **I have redeployed the build and updated the cache-control header**, and I am still facing the same issue.

I can't figure out where the caching is happening. If it is at CloudFront, how do I manage this? I don't have access to the CloudFront distribution; Amplify hides that.

Here is the call I am making:
```
	const clientReviewsQueryResults = await (
		API.graphql({
			query: listClientReviewModels,
			authMode: 'API_KEY',
			variables: { limit: 10, filter: reviewFilter}
		})
	);
```

Here is my NextJS Config file:
```
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  swcMinify: true,
}
const withBundleAnalyzer = require("@next/bundle-analyzer")({
  enabled: process.env.ANALYZE === "true",
});

module.exports = withBundleAnalyzer({
  ...nextConfig,
  async redirects() {
    return [
      {
        source: '/blog',
        destination: '/',
        permanent: true,
      },
      {
        source: '/contact',
        destination: '/',
        permanent: true,
      },
    ]
  },
})
```

Here are the HTTP Response Headers I get:

**Response**
```
:status: 200
Cache-Control: private, no-cache, no-store, max-age=0, must-revalidate
Content-Encoding: gzip
Content-Type: text/html; charset=utf-8
Date: Tue, 13 Jun 2023 14:25:53 GMT
Server: CloudFront
Vary: Accept-Encoding
Via: 1.1 9adef5b1c5fc9ca80d6f4f8d19e103a2.cloudfront.net (CloudFront)
x-amz-cf-id: N35s80oU9t24Tq-gfsywrlMfahxARl_kJKBK58PBjoU4c8Fytu6Bkw==
x-amz-cf-pop: SEA19-C3
x-cache: Miss from cloudfront
x-powered-by: Next.js
```
In summary - I would like to be able to NOT cache calls made to the backend (API/AppSync/etc) when using `API.query({...})`

**Comment by hloriii** [comment]
Hi @essedbl 👋🏽 thanks for raising this issue and providing these details. According the response headers, the response seems to not come from the CDN cache which is indicated by `x-cache: Miss from cloudfront`. Also, the `Cache-Control` settings appear to have `max-age=0` which would tell the client to update the data immediately so that may rule out Next.js caching this data as well. 

I'm working with the team to pinpoint where the caching is happening and will get back to you with next steps. Thank you for your patience.

**Comment by essedbl** [comment]
Hi there. I figured out what was going on. When I delete a record in the Amplify console, in the Content section, it doesn’t actually delete the data in the underlying DynamoDB table. Rather, it sets a “_deleted” field to true, and sets a ttl on the record to 30 days. The Amplify console ignores the record, however, the API.query function does not ignore it. Further, the _deleted field is not defined in the GraphQL schema, and thus the field cannot be used in a filter expression.

My workload is such that I can manually delete the record from DynamoDB so at this point it is a small inconvenience for me. However, I think it would make sense to have consistent behavior between the Amplify console and the API category’s capabilities.

Thanks!

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by hloriii** [comment]
Hi @essedbl Thanks for updating us. I can see how that inconsistency can cause confusion. I will take this feedback to the team for future improvements.

**Comment by chrisbonifacio** [comment]
Hi @essedbl , I see you already understand the cause of the described issue. This is intended behavior, but the way you can filter out the soft deleted records in your graphql requests is to simply add the `_deleted: Boolean` field to your model in the schema and it will be available in the filter fields upon push.

**Comment by chrisbonifacio** [comment]
@essedbl it turns out there was a PR merged not too long ago that exposes the _deleted field so it can be used without explicitly including it on the model. 

https://github.com/aws-amplify/amplify-category-api/pull/1382

Please make sure you are on the latest version of the Amplify CLI to take advantage of this.
